File: src/main/java/de/cloudypanda/main/integrations/WebhookManager.java
package de.cloudypanda.main.integrations;

import de.cloudypanda.main.Huntcraft;

import javax.net.ssl.HttpsURLConnection;
import java.io.IOException;
import java.io.OutputStream;
import java.net.*;

public class WebhookManager {

    private static final String webhookUrl = "[WebHook]";

    public static void sendDeathMessage(String deathMessage) {
        String requestContent = String.format(
                """
                        {
                            "username": "Death Bot",
                            "avatar_url":"https://www.iconpacks.net/icons/1/free-trash-icon-347-thumb.png",
                            "content": "@here %s"
                        }
                """, deathMessage);
        try {
            URL url = new URL(webhookUrl);
            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
            connection.addRequestProperty("Content-Type", "application/json");
            connection.setDoOutput(true);
            connection.setRequestMethod("POST");

            OutputStream stream = connection.getOutputStream();
            stream.write(requestContent.getBytes());
            stream.flush();
            stream.close();

            connection.getInputStream().close(); //I'm not sure why but it doesn't work without getting the InputStream
            connection.disconnect();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/java/de/cloudypanda/main/core/integrations/discord/WebhookManager.kt
package de.cloudypanda.main.core.integrations.discord

import de.cloudypanda.main.Huntcraft
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody


class WebhookManager {

    companion object {
        private var client: OkHttpClient = OkHttpClient()

        fun sendDeathMessage(deathMessage: String) {
            val coreConfigModel = Huntcraft.instance.coreConfigManager.readFromFile()
            if(!coreConfigModel.webhook.enabled) return

            val requestContent = String.format(
                """
                        {
                            "username": "Death Bot",
                            "avatar_url":"https://www.iconpacks.net/icons/1/free-trash-icon-347-thumb.png",
                            "content": "@here %s"
                        }
                """, deathMessage
            )
            val request = Request.Builder()
                .url(coreConfigModel.webhook.webhookUrl)
                .post(requestContent.toRequestBody())
                .header("Content-Type", "application/json")
                .build()

            client.newCall(request).execute().use { response ->
                println(response.body!!.string())
            }
        }

        fun sendAchievementMessage(message: String) {
            val coreConfigModel = Huntcraft.instance.coreConfigManager.readFromFile()
            if(!coreConfigModel.webhook.enabled) return

            val requestContent = String.format(
                """
                        {
                            "username": "Achievement Bot",
                            "avatar_url":"https://www.iconpacks.net/free-icon/medal-1369.html",
                            "content": "@here %s"
                        }
                """, message
            )
            val request = Request.Builder()
                .url(coreConfigModel.webhook.webhookUrl)
                .post(requestContent.toRequestBody())
                .header("Content-Type", "application/json")
                .build()

            client.newCall(request).execute().use { response ->
                println(response.body!!.string())
            }
        }
    }
}
